From ee3a7de0882ec8317c1eacc12c706b7156809f50 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Mon, 21 May 2018 15:15:37 +0300
Subject: [PATCH 3831/5242] MLK-18358: drm/panel: rm67191: Fix initial
 brightness level

commit  566255d5a94194829422486d91d89cecfa25fc26 from
https://source.codeaurora.org/external/imx/linux-imx.git

When enabling the panel, the initial brightness level was hard-coded to
0x20. This way, during a suspend/resume cycle, after resume, this
hard-coded brightness was used, instead of the one before suspend.
Removing the hard-coded level and using the one stored in backlight
device.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/panel/panel-raydium-rm67191.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/panel-raydium-rm67191.c b/drivers/gpu/drm/panel/panel-raydium-rm67191.c
index 3be2412..7cd1bf5 100644
--- a/drivers/gpu/drm/panel/panel-raydium-rm67191.c
+++ b/drivers/gpu/drm/panel/panel-raydium-rm67191.c
@@ -284,6 +284,7 @@ static int rad_panel_enable(struct drm_panel *panel)
 	struct mipi_dsi_device *dsi = rad->dsi;
 	struct device *dev = &dsi->dev;
 	int color_format = color_format_from_dsi_format(dsi->format);
+	u16 brightness;
 	int ret;
 
 	if (rad->enabled)
@@ -343,7 +344,8 @@ static int rad_panel_enable(struct drm_panel *panel)
 		goto fail;
 	}
 	/* Set display brightness */
-	ret = mipi_dsi_dcs_set_display_brightness(dsi, 0x20);
+	brightness = rad->backlight->props.brightness;
+	ret = mipi_dsi_dcs_set_display_brightness(dsi, brightness);
 	if (ret < 0) {
 		DRM_DEV_ERROR(dev, "Failed to set display brightness (%d)\n",
 			      ret);
-- 
1.7.9.5

